tools/mtv-api-tests.py

```
@@ -0,0 +1,1177 @@
+
+    Returns:
+        Connected vSphere ServiceInstance.
+
+    Raises:
+        ConnectionError: If connection fails.
+    """
+    with console.status("[bold]Connecting to vSphere..."):
+        try:
+            si = SmartConnect(host=host, user=username, pwd=password, port=443, disableSslCertValidation=True)
+        except vim.fault.InvalidLogin:
+            raise ConnectionError(f"Invalid vSphere credentials for host '{host}'")
```

Ruff (0.15.4)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)